### ISSUE TITLE
Unexpected token svelte(parse-error)

### ISSUE THREAD

**Issue by CAJazzer** [UnwrittenFun/svelte-vscode#39]
Sorry if a duplicate.  Trying to use Svelte language support in VScode with TypeScript.  First variable to use a type will display an editor error.
```
<p>Pretty Boring Component</p>
<p>{ withoutACast }</p>
<p>{ causeAnError }</p>
<p>{ amCool }</p>

<script lang="ts">
  const withoutACast = 'I have no cast so I am cool';
  const causeAnError: string = 'I cause an editor error';
  const amCool: string = 'for some reason I do not cause an error';
</script>
```

**Comment by r0skar** [comment]
Just wanted to add, that I am getting the same error using  https://github.com/pyoner/svelte-ts-template.

**Comment by r0skar** [comment]
Sorry for my quick response yesterday without reading through existing issues. After taking the time to do so, it seems that this plugin requires a `svelte.config.js` file when you want to use preprocessors (TS, SASS etc.). See https://github.com/UnwrittenFun/svelte-vscode/issues/28#issuecomment-458071004.

Now, I dont know what kind of setup you run @CAJazzer, but I am using https://github.com/pyoner/svelte-ts-preprocess. In order to get rid of the parse-errors, you want to create a `svelte.config.js` file like this:

```
const { preprocess } = require("@pyoner/svelte-ts-preprocess")

module.exports = {
  preprocess: preprocess()
}
```

Reload VSCode and the errors should be gone.

**Comment by CAJazzer** [comment]
Also running [https://github.com/pyoner/svelte-ts-preprocess](https://github.com/pyoner/svelte-ts-preprocess).  Yes, this resolved the issue.  Saw a mention of svelte.config.js but did not know how to construct the contents.  Thank you.

**Comment by laptou** [comment]
The requirement of `svelte.config.js` should be properly documented ... I spent 2 days looking for this information :(

**Comment by laptou** [comment]
`svelte.config.js` is not working for me. Here is my file:

```js
const preprocess = require("svelte-preprocess");

module.exports = {
  preprocess: preprocess()
};
```

And here is the output when I open a bare-bones `.svelte` file in VSCode:

```
Using Svelte v3.4.1 from /home/ibi/projects/svelte-widgets/node_modules/svelte/compiler
(node:6430) UnhandledPromiseRejectionWarning: Error: [svelte-preprocess] Error transforming 'typescript'. Message:
The "path" argument must be of type string. Received type undefined
    at throwError (/home/ibi/projects/svelte-widgets/node_modules/svelte-preprocess/src/utils.js:14:9)
    at exports.runTransformer (/home/ibi/projects/svelte-widgets/node_modules/svelte-preprocess/src/utils.js:78:5)
    at Object.script (/home/ibi/projects/svelte-widgets/node_modules/svelte-preprocess/src/autoProcess.js:77:38)
    at /home/ibi/.vscode/extensions/jamesbirtles.svelte-vscode-0.9.1/node_modules/svelte-language-server/dist/src/plugins/SveltePlugin.js:192:64
    at step (/home/ibi/.vscode/extensions/jamesbirtles.svelte-vscode-0.9.1/node_modules/svelte-language-server/dist/src/plugins/SveltePlugin.js:43:23)
    at Object.next (/home/ibi/.vscode/extensions/jamesbirtles.svelte-vscode-0.9.1/node_modules/svelte-language-server/dist/src/plugins/SveltePlugin.js:24:53)
    at /home/ibi/.vscode/extensions/jamesbirtles.svelte-vscode-0.9.1/node_modules/svelte-language-server/dist/src/plugins/SveltePlugin.js:18:71
    at new Promise (<anonymous>)
    at __awaiter (/home/ibi/.vscode/extensions/jamesbirtles.svelte-vscode-0.9.1/node_modules/svelte-language-server/dist/src/plugins/SveltePlugin.js:14:12)
    at preprocessor.script (/home/ibi/.vscode/extensions/jamesbirtles.svelte-vscode-0.9.1/node_modules/svelte-language-server/dist/src/plugins/SveltePlugin.js:188:56)
(node:6430) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:6430) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This is the `.svelte` file that I am looking at:

```html
{#if r}
hello
{/if}

<script lang="ts">
let r: number = 3; 
</script>
```

**Comment by laptou** [comment]
The IDE doesn't register any errors except inside of the log (there are no errors in the actual editors, which is an improvement from when I didn't have `svelte.config.js` at all) but there is no Intellisense in the file.

**Comment by CAJazzer** [comment]
The ts preprocessor plugin reference has changed (at least for me using rollup).  If you are building a new project with the pyoner/svelte-ts-template  you should create a svelte.config.js that looks like this:
```
const { preprocess } = require("svelte-ts-preprocess")

module.exports = {
  preprocess: preprocess()
}
```

This worked for me.

**Comment by hmmhmmhm** [comment]
> Sorry for my quick response yesterday without reading through existing issues. After taking the time to do so, it seems that this plugin requires a `svelte.config.js` file when you want to use preprocessors (TS, SASS etc.). See [#28 (comment)](https://github.com/UnwrittenFun/svelte-vscode/issues/28#issuecomment-458071004).
> 
> Now, I dont know what kind of setup you run @CAJazzer, but I am using https://github.com/pyoner/svelte-ts-preprocess. In order to get rid of the parse-errors, you want to create a `svelte.config.js` file like this:
> 
> ```
> const { preprocess } = require("@pyoner/svelte-ts-preprocess")
> 
> module.exports = {
>   preprocess: preprocess()
> }
> ```
> 
> Reload VSCode and the errors should be gone.

Thanks! I finally found it!

But Typescript Intelligence was paralyzed.... It would be great if we had intelligence. Is there any other way?

**Comment by clintwood** [comment]
@laptou are you still getting the `UnhandledPromiseRejectionWarning` issue?

@UnwrittenFun, I'm experiencing the same issue and from a little digging I see that `filename` is undefined in the call to [`script`](https://github.com/pyoner/svelte-typescript/blob/master/packages/preprocess/src/svelte-ts-preprocess.ts#L162).

Is this a config issue, an API issue or just a bug?

[EDIT]
To be clear I am using the suggested  `svelte.config.js` as mentioned above:
```js
const { preprocess } = require("@pyoner/svelte-ts-preprocess")

module.exports = {
  preprocess: preprocess()
}
```
Svelte v3.6.6
@pyoner/svelte-ts-preprocess v1.2.0

**Comment by clintwood** [comment]
@UnwrittenFun this would be resolved with PR UnwrittenFun/svelte-language-server#18 - until then `svelte-vscode` is broken!

**Comment by jamesbirtles** [comment]
should be fixed in v0.9.2
